### ISSUE TITLE
Relative positions, or Alibi, or attention_vias (T5) support

### ISSUE THREAD

**Issue by vince62s** [Dao-AILab/flash-attention#332]
Hello,
At the moment it is impossible to use flash attention with one of those 3 above because it requires to alter the scores before the softmax.

Each of the 3 is a different way to encode positions but in the end for Relative positions or attention boas (T5) it is just a matter of adding a term to scores before the softmax and in the case of Alibi it's just applying a function.

Would there be an easy way to pass a function within flash_attn() so that it takes this into account and hence support those ?

Cheers. Great library btw.

**Comment by tridao** [comment]
Right, none of those are supported right now. Eventually they should be supported, I just don't know when as there are a few other priorities.

**Comment by cccc0der** [comment]
> Right, none of those are supported right now. Eventually they should be supported, I just don't know when as there are a few other priorities.

Wish bias too :)
